code/game/objects/structures/holosign.dm

```diff
@@ -8,22 +8,49 @@
 	max_integrity = 1
 	armor = list("melee" = 0, "bullet" = 50, "laser" = 50, "energy" = 50, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 20, "acid" = 20)
 	var/obj/item/holosign_creator/projector
+	var/lifespan = 5 MINUTES
+	var/death_time
+	var/infinite = FALSE //for infinite holosigns
+
+	var/countdown_color
+	var/obj/effect/countdown/holosign/countdown
 
 /obj/structure/holosign/New(loc, source_projector)
 	if(source_projector)
 		projector = source_projector
 		projector.signs += src
 	..()
 
-/obj/structure/holosign/Initialize()
+/obj/structure/holosign/Initialize(mapload, source, new_lifespan)
 	. = ..()
 	alpha = 0
 	SSvis_overlays.add_vis_overlay(src, icon, icon_state, ABOVE_MOB_LAYER, plane, dir, add_appearance_flags = RESET_ALPHA) //you see mobs under it, but you hit them like they are above it
 
+	if(!infinite)
+		START_PROCESSING(SSobj, src)
+
+		if(new_lifespan)
+			lifespan = new_lifespan
+		death_time = world.time + lifespan
+		countdown = new(src)
+		if(countdown_color)
+			countdown.color = countdown_color
+		countdown.start()
+
+/obj/structure/holosign/process()
+	if(death_time < (world.time + 60 SECONDS) && death_time > (world.time + 58 SECONDS)) //there has to be a better way to get this to just trigger once
```

Comment:
any reason not to do something like 
if(countdown.invisibility && death_time < (world.time + 60 SECONDS))
instead?